qt/mainwindow.cpp

```diff
@@ -578,6 +662,24 @@ void MainWindow::CreatePanelImpl(size_t i, Qt::DockWidgetArea area, QString cons
   }
 }
 
+void MainWindow::CreateTrafficPanel(string const & dataFilePath, string const & sampleFilePath)
+{
+  CreatePanelImpl(1, Qt::RightDockWidgetArea, tr("Traffic"), QKeySequence(), nullptr);
+
+  m_trafficMode = new TrafficMode(dataFilePath, sampleFilePath,
+                                     m_pDrawWidget->GetFramework().GetIndex(),
+                                     make_unique<TrafficDrawerDelegate>(m_pDrawWidget));
+  m_Docks[1]->setWidget(new TrafficPanel(m_trafficMode, m_Docks[1]));
+  m_Docks[2]->adjustSize();
```

Comment:
Are you sure? The size of m_Docks is 2, so valid indices are 0 and 1 only, isn't it?